File: Тренировка 1.0/03. Множества/E. OpenCalculator/OpenCalculator.cs
using System;
using System.Collections.Generic;

string[] xyz = Console.ReadLine().Split(' ');

HashSet<int> set1 = new HashSet<int>() { int.Parse(xyz[0]), int.Parse(xyz[1]), int.Parse(xyz[2]) };
HashSet<int> set2 = new HashSet<int>();
int number = int.Parse(Console.ReadLine());
while(number != 0)
{
    set2.Add(number % 10);
    number /= 10;
}

set2.ExceptWith(set1);

Console.WriteLine(set2.Count);

File: Тренировка 1.0/03. Множества/E. OpenCalculator/OpenCalculator.cpp
#include <set>
#include <vector>
#include <iostream>
#include <algorithm>

int main()
{
    char x, y, z;
    std::cin >> x >> y >> z;

    std::set<char> set1 = {x, y, z};
    std::set<char> set2;
    std::string number;
    std::cin >> number;
    for (int i = 0, n = number.size(); i < n; ++i) set2.emplace(number[i]);

    std::vector<int> missingDigits;
    std::set_difference(set2.begin(), set2.end(), set1.begin(), set1.end(), std::back_inserter(missingDigits));
    std::cout << missingDigits.size();
}
